### ISSUE TITLE
R Session crashes everytime with LightGBM Under Mac OSX

### ISSUE THREAD

**Issue by danielmarcelino** [microsoft/LightGBM#2267]
# The Environment info

Operating System: OSX

CPU/GPU model: Macbook Pro 

C++/Python/R version: R 3.6.1

LightGBM version or commit hash: 36c89134d4013b2e5e45bc55656a18bd6141995a

## Error message

My R crashes every time I try to train a model. The installation of LightGBM was run (apparently) successful, but so far I never could fit a single model. My R always crashed. How can I figure out the likely cause and fix it?  
 
![installation](https://user-images.githubusercontent.com/2594275/61314076-265b5000-a7d2-11e9-87b3-acfa9d29d299.png)

![R screen](https://user-images.githubusercontent.com/2594275/61314017-0592fa80-a7d2-11e9-97f8-25794f955734.png)

## Reproducible examples

```
require(lightgbm)
require(methods)

# We load in the agaricus dataset
# In this example, we are aiming to predict whether a mushroom is edible
data(agaricus.train, package = "lightgbm")
data(agaricus.test, package = "lightgbm")
train <- agaricus.train
test <- agaricus.test
```

```
# The loaded data is stored in sparseMatrix, and label is a numeric vector in {0,1}
class(train$label)
class(train$data)


print("Training lightgbm with sparseMatrix")
bst <- lightgbm(data = train$data,
                label = train$label,
                num_leaves = 4,
                learning_rate = 1,
                nrounds = 2,
                objective = "binary")
```

**Comment by jameslamb** [comment]
Thanks for reporting this! I can help work through it. I've never observed this before with the R package, but we have seen it on the Python side (see `Question 10` in [the LightGBM FAQs](https://lightgbm.readthedocs.io/en/latest/FAQ.html)).

Are you using `gcc` to compile `LightGBM`? `gcc` comes with `libomp` bundled already, so if you are using that AND did something like `brew install libomp`, it can lead to this conflict at run time.

One fix suggested in dmlc/xgboost#1715, specifically [this comment](https://github.com/dmlc/xgboost/issues/1715#issuecomment-313308100):, is to remove the version(s) of `libomp` known to Homebrew.

```
brew uninstall libiomp clang-omp
```

Could you try that, then reinstall `LightGBM`, and let me know if it works for you? If that works I will add it to our FAQs.

**Comment by jameslamb** [comment]
@danielmarcelino are you still facing this issue? I'd be happy to help if you can answer the questions in my comment above.

Thanks!

**Comment by jameslamb** [comment]
No problem! We are here to help YOU, just want to know if it is still an issue which other people might be facing.

I'm glad you got it working on Linux, but I am worried we have some problem on Mac OS (see #2350 for example). Could you tell me _how_ you are installing `lightgbm` on your Mac? Like show me the exact code you are running? Namely, I'm curious about whether you are using `gcc` or `clang`.

**Comment by StrikerRUS** [comment]
Ah! Only after a month since creating the issue I managed to take a closer look at the provided screenshot. And seems that I found the root cause of the crash.

Refer to https://lightgbm.readthedocs.io/en/latest/FAQ.html#lightgbm-crashes-randomly-with-the-error-like-initializing-libiomp5-dylib-but-found-libomp-dylib-already-initialized.

![image](https://user-images.githubusercontent.com/25141164/63929416-0918d300-ca5a-11e9-8c18-1fd02ce3d51b.png)
